### DISCUSSION TITLE
Plans for additional little flourishes

### DISCUSSION THREAD

**Discussion by pserwylo** [beat-feet/beat-feet#41] [Ideas]
A few of the little things I've done so far include:
* Shaking the camera after hitting an obstacle (more for larger obstacles).
* Adding extra details to the fence sprites (random bushes).

Some thoughts of other things to add:
* Dust or blood splatter after hitting objects.
* A little flourish when jumping, and another when double jumping, e.g. a feint rainbow, similar to the one on the icon.
* Vibration after hitting an obstacle

I'm sure there could be more, worth documenting here.

**Comment by pserwylo** [comment]
Since originally wrigint this, the extra flourishes I've added include:
* Blood particles dropping out of hearts when losing health.
* Rainbow particles ejecting from the character at an ever increasing rate as combos increase (I'd like to tie this to some sort of mechanics beyond just graphics soon - e.g. a bar of rainbow "stuff" that can be used for... something? Perhaps extra jumping height or something like that).
* Increased combo's are animated to draw attention to them
* Vibration on android devices after hitting objects, commensurate with the size of the object.

**Comment by r0ckyfeller** [comment]
Just tested out v0.1.0, impatiently awaiting v0.2.0. I've been waiting for this concept to be realized so thank you so much for being the one to do it. As for the flourishes, I'd think instead of a black back drop you could add a starry night sky in pixel art to fit the style, or a cloudy sky of you want it to be day. Keep up the great work I can't wait to keep playing (even though I can't make it past 9% :/ )

**Reply by pserwylo** [reply]
Thanks for the kind words. I've also been thinking about this concept (admittedly only intermittedly) since I was a kid and saw (but never played) this game: https://en.wikipedia.org/wiki/Vib-Ribbon.

As for the background, agreed. I think that is quite an important graphical element to add. I'd love even more if there was a way to tie it to the music somehow, but perhaps just start with *something* and then tie it to the music later.

I would like it to be at least proceduraly generated tough (even if random) so that it can keep up with the whole "the level looks different each time" thing.

By the way, 9% is a fine first effort!

**Comment by Sogolumbo** [comment]
The vibration is a nice touch but can be annoying in specific situations or with specific devices.
I'd suggest to also add a visual indicator for losing life/protection (either blood, as you proposed or maybe a rainbow border around the whole screen for losing protection and a red one for losing life points).
When the visual indicator is there, please add a setting which allows removing the vibrations.
